packages/react/src/components/TextInput/TextInput.stories.js

```diff
@@ -28,154 +27,220 @@ export default {
   subcomponents: {
     TextInputSkeleton,
   },
+  args: {
+    className: 'input-test-class',
+    id: 'text-input-1',
+    placeholder: 'Placeholder text',
+    invalid: false,
+    invalidText: 'Error message goes here',
+    disabled: false,
+    labelText: 'Label text',
+    helperText: 'Helper text',
+    warn: false,
+    warnText:
+      'Warning message that is really long can wrap to more lines but should not be excessively long.',
+    size: 'md',
+    readOnly: false,
+    inline: false,
+    hideLabel: false,
+    enableCounter: false,
+    maxCount: 10,
+    type: 'text',
+    defaultWidth: 300,
+    defaultValue: '',
+  },
   argTypes: {
-    light: {
-      table: {
-        disable: true,
+    className: {
+      control: {
+        type: 'text',
       },
     },
-    slug: {
-      table: {
-        disable: true,
+    defaultValue: {
+      control: {
+        type: 'text',
       },
     },
-  },
-};
-
-const sharedArgTypes = {
-  defaultWidth: {
-    control: { type: 'range', min: 300, max: 800, step: 50 },
-  },
-  className: {
-    control: {
-      type: 'text',
+    placeholder: {
+      control: {
+        type: 'text',
+      },
     },
-  },
-  defaultValue: {
-    control: {
-      type: 'text',
+    invalid: {
+      control: {
+        type: 'boolean',
+      },
     },
-  },
-  placeholder: {
-    control: {
-      type: 'text',
+    invalidText: {
+      control: {
+        type: 'text',
+      },
     },
-  },
-  invalid: {
-    control: {
-      type: 'boolean',
+    disabled: {
+      control: {
+        type: 'boolean',
+      },
     },
-  },
-  invalidText: {
-    control: {
-      type: 'text',
+    labelText: {
+      control: {
+        type: 'text',
+      },
     },
-  },
-  disabled: {
-    control: {
-      type: 'boolean',
+    helperText: {
+      control: {
+        type: 'text',
+      },
     },
-  },
-  labelText: {
-    control: {
-      type: 'text',
+    warn: {
+      control: {
+        type: 'boolean',
+      },
     },
-  },
-  helperText: {
-    control: {
-      type: 'text',
+    warnText: {
+      control: {
+        type: 'text',
+      },
     },
-  },
-  warn: {
-    control: {
-      type: 'boolean',
+    value: {
+      control: {
+        type: 'text',
+      },
     },
-  },
-  warnText: {
-    control: {
-      type: 'text',
+    onChange: {
+      action: 'onChange',
     },
-  },
-  value: {
-    control: {
-      type: 'text',
+    onClick: {
+      action: 'onClick',
     },
-  },
-  onChange: {
-    action: 'onChange',
-  },
-  onClick: {
-    action: 'onClick',
-  },
-  size: {
-    options: ['sm', 'md', 'lg', 'xl'],
-    control: {
-      type: 'select',
+    size: {
+      options: ['sm', 'md', 'lg', 'xl'],
```

Comment:
```suggestion
      options: ['sm', 'md', 'lg'],
```

`xl` control doesn't work as there's no `xl` variant for this component so lets just remove it.